### PR TITLE
Preallocate result array for range

### PR DESCRIPTION
**Pull request by jacereda**
For `range 0 1000`:

No prealloc:
```
   range/1000           6020.450 ns 0.998 r2       166100.536 ops/s 

```
Prealloc:

```
   range/1000           3720.389 ns 0.995 r2       268789.072 ops/s 
```

### CODE REVIEW

**Comment by jacereda** [comment]
Ups, I always forget `range` is inclusive...

**Comment by jacereda** [comment]
With the correct size:

```
   range/1000           2580.780 ns 0.998 r2       387479.709 ops/s 
```

File: src/Data/Array.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 exports.range = function (start) {
   return function (end) {
     var step = start > end ? -1 : 1;
-    var result = [];
+    var result = new Array(step * (end - start) + 1);
```
Comment (paf31) [inline src/Data/Array.js]: This seems wrong. Isn't this too large when `step > 1`?

Comment (jacereda) [reply src/Data/Array.js]: What do you mean? It can only be -1 or 1...

Comment (paf31) [reply src/Data/Array.js]: Oops, sorry, I misread :man_facepalming:

**Comment by paf31** [comment]
Thanks!

**Pull request merged**